### DISCUSSION TITLE
Discussion: Add the header property to all input controls

### DISCUSSION THREAD

**Discussion by niels9001** [microsoft/microsoft-ui-xaml#7495] [General]
# Discussion: Add the header property to all button controls
Most input controls have a header property, that allows developer to add  e.g. a descriptive text.

This is useful, but for the button control family this property is not available. This requires developers to add a separate textblock. This however requires additional styling (in terms of margins, paddings and fontstyles).

This is quite cumbersome, especially when using controls with a header on the same page since things look off and inconsistent across the page.

Controls to consider:
- Button
- CheckBox
- ColorPicker
- DropDownButton
- HyperlinkButton
- RadioButton
- RatingControl
- RepeatButton
- SplitButton
- ToggleButton
- ToggleSplitButton

Some examples where a header property would be useful:

![image](https://user-images.githubusercontent.com/9866362/82730962-f50db380-9d03-11ea-84a0-2834a9f48f5b.png)

![image](https://user-images.githubusercontent.com/9866362/82730997-26867f00-9d04-11ea-83c0-80fc1fc3a7f6.png)

**Comment by mdtauk** [comment]
The proposed form control may be useful for this I'm the future.

**Comment by ranjeshj** [comment]
@niels90001 Would your scenarios be covered by the forms control as @mdtauk mentioned ? The proposal is [here](https://github.com/microsoft/microsoft-ui-xaml/issues/82).

**Comment by niels9001** [comment]
> @niels90001 Would your scenarios be covered by the forms control as @mdtauk mentioned ? The proposal is [here](https://github.com/microsoft/microsoft-ui-xaml/issues/82).

I guess it would.. it would be nice to have this consistent, for usecases where only 1 control is used.

But yes, I could live with a forms control to solve this issue :)!
